translator/util/ec2util/ec2util.go

```diff
@@ -67,46 +76,54 @@ func initEC2UtilSingleton() (newInstance *ec2Util) {
 		log.Println("W! [EC2] Sleep until network is up")
 		time.Sleep(1 * time.Second)
 	}
+
 	if !networkUp {
 		log.Println("E! [EC2] No available network interface")
 	}
 
-	ses, err := session.NewSession()
+	err := newInstance.getEC2MetadataFromIMDS()
+
 	if err != nil {
-		log.Println("E! [EC2] getting new session info: ", err)
-		return
+		log.Println("E! [EC2] Cannot get EC2 Metadata from IMDS:", err)
 	}
-	md := ec2metadata.New(ses)
 
-	if !md.Available() {
-		log.Println("E! ec2metadata is not available")
-		return
-	}
+	return
+}
 
-	if info, err := md.GetMetadata("instance-id"); err == nil {
-		newInstance.InstanceID = info
-	} else {
-		log.Println("E! getting instance-id from EC2 metadata fail: ", err)
+func (e *ec2Util) getEC2MetadataFromIMDS() error {
```

Comment:
Nitpick - using a `get` verb for this function makes me think that it will return something, but instead you are just setting values in the `ec2Util` struct. I would either change the name of the function to something like `deriveMetadata`, or I would suggest changing the function definition to return the values in a struct, and then the calling function can use the values to set on the ec2Util object itself.

Reply:
Changing the naming convention would be a better idea since it aligns with ECS Util behavior. Thanks for the recommendation. 👍